src/p2p/behaviour.rs

```diff
@@ -496,15 +496,17 @@ impl<Types: IpfsTypes> Behaviour<Types> {
             .create_subscription(self.kademlia.get_closest_peers(id.as_bytes()).into(), None)
     }
 
-    pub fn get_providers(&mut self, key: Key) -> SubscriptionFuture<KadResult, String> {
+    pub fn get_providers(&mut self, cid: Cid) -> SubscriptionFuture<KadResult, String> {
+        let key = Key::from(cid.to_bytes());
         self.kad_subscriptions
             .create_subscription(self.kademlia.get_providers(key).into(), None)
     }
 
     pub fn start_providing(
         &mut self,
-        key: Key,
+        cid: Cid,
     ) -> Result<SubscriptionFuture<KadResult, String>, anyhow::Error> {
+        let key = Key::from(cid.to_bytes());
```

Comment:
it amazes me that this conversion is enough but it probably is, for the interface-http-core tests.

Could there be some comment or similar on why this isn't a multihash, or about the future plans of changing it to multihash or something?

Reply:
yeah, as far as I can remember js-ipfs has not moved on to `Multihash`es yet; I can comment on it

Reply:
I believe 0.49 has, or maybe they're planning on it for 0.50...

Reply:
yeah, plain `Multihash` works here too